docs/evm_vs_eravm.md

```diff
@@ -0,0 +1,309 @@
+# EraVM vs EVM
+
+## Overview
+
+zkSync is a zk-Rollup meant to be fully EVM compatible. In practice, this can mean a number of different things; for zkSync, it means that you can take any Solidity, Yul, or Vyper contract and deploy it using the zkSync hardhat/foundry tools. This seems like full compatibility, but it's not. The EraVM has a completely different architecture than the EVM; compatibility is provided through `zksolc`, a compiler written by Matter Labs that takes any Solidity, Yul or Vyper contract and compiles it down to the EraVM bytecode.
+
+This is an important difference. The following Solidity contract:
+
+```
+contract Test {
+    function main(uint256 a, uint256 b) external pure returns(uint256 result) {
+        result = a + b;
+    }
+}
+```
+
+compiles to an EVM assembly that looks like this:
+
+```
+PUSH1 0x80
+PUSH1 0x40
+MSTORE
+CALLVALUE
+DUP1
+ISZERO
+PUSH1 0xE
+JUMPI
```

Comment:
This assembly is weird, why is there no addition? Is the code for the function or a call site?

Reply:
This is not the complete assembly, almost all compiled solidity contracts start this way, it is basically checking which function to execute

Reply:
OK, but is it not supposed to be the assembly for the sample snippet?

Reply:
@Oppen Ideally it would be, the problem is it's really hard to isolate just the code for the snippet. `solc` generates the entire code for the contract function dispatching, etc.

The idea here was more to show how different both assemblies are (one is a super simple stack based one, the other one a complex register based one) at a high level